src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -1409,12 +1409,43 @@ void InterpreterSelectQuery::executeFetchColumns(
             query_info.input_order_info = query_info.order_optimizer->getInputOrder(storage, metadata_snapshot);
         }
 
-        auto read_step = std::make_unique<ReadFromStorageStep>(
-            table_lock, metadata_snapshot, options, storage,
-                required_columns, query_info, context, processing_stage, max_block_size, max_streams);
+        LocalLimits limits;
+        std::shared_ptr<const EnabledQuota> quota;
 
-        read_step->setStepDescription("Read from " + storage->getName());
-        query_plan.addStep(std::move(read_step));
+        /// Set the limits and quota for reading data, the speed and time of the query.
+        if (!options.ignore_limits)
```

Comment:
I agree that the whole refactoring is straightforward and clear and here it is a much more suitable place for imposing limitations. Also, I am aware, that we tend to write longs functions in Clickhouse. But nevertheless what if we refactor the code below out into a standalone function?